api4/post.go

```diff
@@ -396,6 +397,54 @@ func getPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPostsByIds(c *Context, w http.ResponseWriter, r *http.Request) {
+	postIDs := model.ArrayFromJSON(r.Body)
+
+	if len(postIDs) == 0 {
+		c.SetInvalidParam("post_ids")
+		return
+	}
+
+	postsList, err := c.App.GetPostsByIds(postIDs)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	var posts = []*model.Post{}
+
+	for _, post := range postsList {
+
+		channel, err := c.App.GetChannel(post.ChannelId)
+		if err != nil {
+			c.Err = err
+			return
+		}
+
+		if !c.App.SessionHasPermissionToChannel(*c.AppContext.Session(), channel.Id, model.PermissionReadChannel) {
+			if channel.Type == model.ChannelTypeOpen {
+				if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), channel.TeamId, model.PermissionReadPublicChannel) {
+					continue
+				}
+			} else {
+				continue
+			}
+		}
+
+		post = c.App.PreparePostForClient(post, false, false)
+
+		posts = append(posts, post)
+	}
+
+	b, jsonErr := json.Marshal(posts)
+	if jsonErr != nil {
+		c.Err = model.NewAppError("getPostsByIds", "api.post.search_posts.invalid_body.app_error", nil, jsonErr.Error(), http.StatusBadRequest)
+		return
+	}
+
+	w.Write(b)
```

Comment:
It's better re-written as:

```go
if err := json.NewEncoder(w).Encode(posts); err != nil {
	mlog.Warn("Error while writing response", mlog.Err(err))
}
```

to make it consistent with others.

Reply:
Updated